### ISSUE TITLE
Giving folks who ask "can I work on this?" more agency

### ISSUE THREAD

**Issue by jacobtomlinson** [dask/community#335]
I've noticed a few instances lately where a new contributor stops by an issue and asks "can I work on this?". I feel disheartened whenever I see this on GitHub because I feel like the half-life of that person's interest is probably only an hour or so on average and if someone doesn't get in there quickly and say yes they will not actually pick up the issue.

We discussed this in the bi-weekly maintainer call and @phofl mentioned that [Pandas has some automation](https://pandas.pydata.org/docs/development/contributing.html?highlight=take#where-to-start) to assign issues if someone comments with the word `take`. They also have more in the way of contributor guidelines than we do to make new contributors feel welcome and give them agency.

I would like for us to improve our docs and also consider similar automation. What do folks think?

**Comment by GenevieveBuckley** [comment]
> I feel like the half-life of that person's interest is probably only an hour or so on average

For what it's worth, I think this is a pretty surprising take! A half hour response time does not sound like something anybody would reasonably expect.

If I was asking "can I work on this" it's because I'd like to try it but expect it to take me a few days or weeks to circle back with something to share, and there's no point me starting if someone else is currently working on it or is likely to pick it up and finish in a day or two. Plus, initial expressions of interest with no follow through happen for all sorts of reasons (not enough time, problem was harder than expected, lost interest, etc.), it's not really accurate to attribute that mostly to how quickly someone originally replied.

**Comment by GenevieveBuckley** [comment]
I agree that improving the contributor documentation here could help. Making it clear that the norm is that anybody can pick things up is great. 

I'm not sure an automated bot response would be that useful. But I don't really like bot responses (they can't provide context or advice). I also don't use the github "assigned" feature to track my workflow.

Even so, I'd still expect these messages to be fairly frequent because:
* Plenty of people will do this with communities they are not familiar with, even if they expect the answer is yes. You need some kind of comment on the thread to know that someone might be working on a thing, phrasing it as a question is a polite way to do that.
* The question is also fulfilling a few other roles. It's not purely about permission, it's about asking for context and if anyone might be working on the problem already. It can nudge maintainers into giving useful advice, and maybe offering to answer questions that might come up.

**Comment by jacobtomlinson** [comment]
Thanks @GenevieveBuckley your perspective is really helpful here. 

Perhaps I am overly worried about the response time. My experience is that if you don't say yes within an hour or so they never come back, but as you say there are other factors that could cause this.

I think updating our docs with more useful guidelines is a must-have to help new contributors feel more confident in picking things up. I also wonder if we could make a small automation that gets triggered by a maintainer adding the "help wanted" label to make a bot comment with a little context about how folks should pick things up and point to that documentation.

I think encouraging folks to say "I intent to work on this" instead of "can I work on this?" removes the initial roadblock of getting started. Obviously, if someone else has said "I intend to work on this" in the last couple of days you know it's probably in hand.

**Comment by martindurant** [comment]
I interpret "can I work on this" to mean "is someone [more experienced] else already on it or planning to do it soon", more or less what @GenevieveBuckley said.
